File: src/main/java/org/kane/domain/service/cooking_stage/CookingStageService.java
package org.kane.domain.service.cooking_stage;

import org.kane.database.entity.recipe_recource.CookingStage;
import org.kane.domain.DTO.entityDTO.cooking_stage.CookingStageCreateDTO;
import org.kane.domain.DTO.entityDTO.cooking_stage.CookingStageEditDescDTO;
import org.kane.domain.DTO.entityDTO.cooking_stage.CookingStageShowDTO;

import java.util.List;

public interface CookingStageService {
    CookingStage createCookingStage(CookingStageCreateDTO cookingStageCreateDTO);
    void editCookingStage(CookingStageEditDescDTO cookingStageEditDescDTO);

    List<CookingStageShowDTO> getAllShowDTOByRecipeID(Long recipeID);
}

File: src/app/service/recipe_resource/cooking-stage.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Endpoint} from '../../util/endpoint';
import CookingStageEditDescDTO from '../../DTO/entity_dto/recipe-recource/cooking_stage/cooking-stage-edit-desc.dto';
import {Observable} from 'rxjs';
import CookingStageShowDTO from '../../DTO/entity_dto/recipe-recource/cooking_stage/cooking-stage-show.dto';

@Injectable({
  providedIn: 'root',
})
export default class CookingStageService {
  private readonly http = inject(HttpClient);
  private readonly cookingStageAPI = new Endpoint('stage');

  editCookingStage(stage:CookingStageEditDescDTO):Observable<any>{
    return this.http.patch<any>(
      this.cookingStageAPI.builder().
      points("edit")
        .build(),
      stage);
  }

  getAllShowDTO(id:number):Observable<CookingStageShowDTO[]>{
    return this.http.get<CookingStageShowDTO[]>(
      this.cookingStageAPI.builder()
        .points("all")
        .addParam("id", id.toString())
    .build())
  }

  deleteCookingStage(id:number):Observable<any>{
    return this.http.delete<any>(
      this.cookingStageAPI.builder()
        .points("delete")
        .addParam("id", id.toString())
        .build())
  }
}
